lib/logger/lib/logger.ex

```diff
@@ -19,9 +19,12 @@ defmodule Logger do
     * Formats and truncates messages on the client
       to avoid clogging `Logger` handlers.
 
-    * Alternates between sync and async modes to remain
-      performant when required but also apply back-pressure
-      when under stress.
+    * Provides multiple forms of [overload protection](https://www.erlang.org/doc/apps/kernel/logger_chapter.html#protecting-the-handler-from-overload):
+      * keeps track of its message queue and switches to sync mode to apply
+        back pressure or even drop messages
+      * limits the number of logs emitted defaulting to 500 per second
```

Comment:
I'm unsure what level of detail to provide as defaults may change, however I found it surprising how compartively low this limit is and so I found it worth calling out